lib/src/app.dart

```diff
@@ -106,6 +112,56 @@ class _AppState extends ConsumerState<Application> {
       setOptimalDisplayMode();
     }
 
+    ref.read(localNotificationServiceProvider).init();
+
+    ref.listenManual(challengesProvider, (prev, current) {
+      final prevIds = prev!.value!.inward.map((challenge) => challenge.id);
+      final inward = current.value!.inward;
+      final repo = ref.read(challengeRepositoryProvider);
+      final l10n = ref.read(l10nProvider).strings;
+
+      inward
+          .map((element) => element.id)
+          .where((id) => !prevIds.contains(id))
+          .map((id) => inward.firstWhere((element) => element.id == id))
+          .forEach((challenge) {
+        ref.read(localNotificationServiceProvider).show(
+              ChallengeNotification(
+                challenge,
+                l10n,
+                onPressed: (action, id) async {
+                  switch (action) {
+                    case ChallengeNotificationAction
+                          .accept: // accept the game and open board
+                      await repo.accept(challenge.id);
+                      final fullId = await repo.show(challenge.id).then(
+                            (challenge) => challenge.gameFullId,
+                          );
+                      pushPlatformRoute(
+                        ref.read(currentNavigatorKeyProvider).currentContext!,
```

Comment:
I wonder if that is necessary to give the current navigator context. I suppose simply passing `context` is enough?